src/ArcgisShare/ArcgisShare.js

```diff
@@ -143,33 +143,42 @@ class ArcgisShare extends Component {
       publicLabel,
       groupsLabel,
       noGroupsLabel,
+      hidePublicSharing,
+      hideOrganizationSharing,
+      hideGroupsSharing,
       portal,
       user
     } = this.props;
 
     return (
       <StyledArcgisShare>
-        <Checkbox
-          labelStyle={{ ...PrimaryCheckboxLabelStyles }}
-          checked={this.state.public || false}
-          onChange={this.publicChange}
-        >
-          {publicLabel}
-        </Checkbox>
-        <Checkbox
-          labelStyle={{ ...PrimaryCheckboxLabelStyles }}
-          checked={this.state.org || this.state.public || false}
-          onChange={this.orgChange}
-          disabled={this.state.public || false}
-        >
-          {portal && portal.name}
-        </Checkbox>
-        <StyledGroupFieldset name="shareGroups">
-          <StyledLegend>{groupsLabel}:</StyledLegend>
-          <StyledGroupContainer>
-            {user && this.getGroupCheckboxes(user.groups, noGroupsLabel)}
-          </StyledGroupContainer>
-        </StyledGroupFieldset>
+        {!hidePublicSharing ? (
+          <Checkbox
+            labelStyle={{ ...PrimaryCheckboxLabelStyles }}
+            checked={this.state.public || false}
+            onChange={this.publicChange}
+          >
+            {publicLabel}
+          </Checkbox>
+        ) : null}
+        {!hideOrganizationSharing ? (
+          <Checkbox
+            labelStyle={{ ...PrimaryCheckboxLabelStyles }}
+            checked={this.state.org || this.state.public || false}
+            onChange={this.orgChange}
+            disabled={this.state.public || false}
+          >
+            {portal && portal.name}
+          </Checkbox>
+        ) : null}
+        {!hideGroupsSharing ? (
+          <StyledGroupFieldset name="shareGroups">
+            <StyledLegend>{groupsLabel}:</StyledLegend>
+            <StyledGroupContainer>
+              {user && this.getGroupCheckboxes(user.groups, noGroupsLabel)}
+            </StyledGroupContainer>
+          </StyledGroupFieldset>
+        ) : null}
```

Comment:
What are your thoughts on doing something more like how the JS API uses [visibleElements](https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-Bookmarks.html#visibleElements) for widgets?